### PR TITLE
[21137] Adapt to RTPS WriterHistory refactor

### PR DESCRIPTION
**Pull request by MiguelCompany**
## Description

This PR adapts the RTPS example code for the refactor in `WriterHistory`.

Related implementation PR:
* eProsima/Fast-DDS#4966

## Contributor Checklist

- [x] Commit messages follow the project guidelines. 
- [x] Code snippets related to the added documentation have been provided.
- [x] Documentation tests pass locally.
- _N/A_ Applicable backports have been included in the description.

## Reviewer Checklist

- [x] The PR has a milestone assigned.
- [x] The title and description correctly express the PR's purpose.
- [x] Check contributor checklist is correct.
- [ ] CI passes without warnings or errors.

### CODE REVIEW

**Comment by MiguelCompany** [comment]
@richiprosima Please test_3 docs

File: code/CodeTester.cpp [hunk 1]
```diff
@@ -151,11 +151,8 @@ void rtps_api_example_create_entities()
     }
 
     //RTPS_API_WRITE_SAMPLE
-    //Request a change from the writer
-    CacheChange_t* change = writer->new_change([]() -> uint32_t
-                    {
-                        return 255;
-                    }, ALIVE);
+    //Request a change from the history
```
Comment (EduPonz) [inline code/CodeTester.cpp:154]: NIT: can you please add a space after `//` in this snippet's comments?

**Review by EduPonz — CHANGES_REQUESTED** [review]
NIT

**Comment by EduPonz** [comment]
@richiprosima Please test_3 docs

**Pull request merged**